### PR TITLE
Sound

### PR DESCRIPTION
**Pull request by Anteanten**

### CODE REVIEW

File: GameProject/Engine/Sound/SoundManager.cpp [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+	alGetListener3f(AL_ORIENTATION, &orientation.x, &orientation.y, &orientation.z);
+	return orientation;
+}
+
+void SoundManager::setMasterVolume(float volume)
+{
+	this->masterVolume = volume;
+	for (unsigned int i = 0; i < sounds.size(); i++) {
+		switch (sounds[i]->getSoundType()) {
+		case(SOUND_MUSIC): sounds[i]->updateSound(masterVolume * musicVolume); break;
```
Comment (TheoBerlin) [inline GameProject/Engine/Sound/SoundManager.cpp]: Add proper indentation for each `case`. Do not place `break` on the same line as the first semicolon.

File: GameProject/Engine/Sound/SoundManager.h [hunk 2]
```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include "Sound.h"
+#include <vector>
+
+class SoundManager {
+private:
+	std::vector<Sound*> sounds;
+
+	//Volume between 0.0 and 1.0
```
Comment (TheoBerlin) [inline GameProject/Engine/Sound/SoundManager.h]: The set functions for volumes should clamp the volumes to [0.0f, 1.0f]

File: GameProject/Game/Components/Explosion.cpp [hunk 3]
```diff
@@ -23,6 +25,8 @@ void Explosion::update(const float & dt)
 		debri->emitter->setPosition(debri->position);
 	}
 
+	sound.setPosition(getHost()->getTransform()->getPosition());
```
Comment (TheoBerlin) [inline GameProject/Game/Components/Explosion.cpp]:
```suggestion
	sound.setPosition(host->getTransform()->getPosition());
```

File: GameProject/main.cpp [hunk 4]
```diff
@@ -23,12 +24,14 @@ int main() {
...
-	//sound.setVolume(0.1f);
-	//sound.playSound();
+	Sound sound;
+	sound.loadSound("Game/assets/sound/dream_catcherMono.wav");
+	sound.setLoopState(true);
+	sound.setPitch(1.25f);
+	sound.playSound();
+	SoundManager::get().addSound(&sound, SOUND_MUSIC);
+
+	sound.stopSound();
```
Comment (TheoBerlin) [inline GameProject/main.cpp]: Do not play the sound if it's going to be stopped immediately after

File: GameProject/Engine/Sound/SoundManager.cpp (hunk 1 above)
Comment (Anteanten) [reply GameProject/Engine/Sound/SoundManager.cpp]: I prefer this method and we have not agreed upon a code standard for this so won't change

**Pull request merged**